src/Libraries/RevitNodes/GeometryConversion/CurveUtils.cs

```diff
@@ -124,6 +149,72 @@ private static bool IsLineLikeInternal(IList<XYZ> points)
             }
         }
 
+        // This test checks end points with directionality. Curves that share end points
+        /// but are reversed are not considered similar.
+        private static bool CurvesHaveSameEndpoints(Curve a, Curve b, double tolerance)
+        {
+            return a.GetEndPoint(0).IsAlmostEqualTo(b.GetEndPoint(0), tolerance) &&
+                a.GetEndPoint(1).IsAlmostEqualTo(b.GetEndPoint(1));
+        }
+
+        private static bool AreSimilar(Line a, Line b, double tolerance)
+        {
+            return CurvesHaveSameEndpoints(a,b, tolerance);
+        }
+
+        private static bool AreSimilar(Arc a, Arc b, double tolerance)
+        {
+            return a.Center.IsAlmostEqualTo(b.Center) && 
+                a.Radius.AlmostEquals(b.Radius, 1.0e-6) && 
+                a.IsCyclic == b.IsCyclic;
+        }
+
+        private static bool AreSimilar(HermiteSpline a, HermiteSpline b, double tolerance)
+        {
+            return CompareRandomParameterLocationDistances(a, b, tolerance);
+        }
+
+        private static bool AreSimilar(NurbSpline a, NurbSpline b, double tolerance)
+        {
+            return CompareRandomParameterLocationDistances(a, b, tolerance);
+        }
+
+        private static bool AreSimilar(Ellipse a, Ellipse b, double tolerance)
+        {
+            return a.Center.IsAlmostEqualTo(b.Center, tolerance) &&
+                a.RadiusX.AlmostEquals(b.RadiusX, tolerance) &&
+                a.RadiusY.AlmostEquals(b.RadiusY, tolerance);
+        }
+
+        private static bool AreSimilar(CylindricalHelix a, CylindricalHelix b, double tolerance)
+        {
+            return a.Height.AlmostEquals(b.Height, tolerance) &&
+                a.BasePoint.IsAlmostEqualTo(b.BasePoint, tolerance) &&
+                a.Pitch.AlmostEquals(b.Pitch, tolerance) &&
+                a.Radius.AlmostEquals(b.Radius, tolerance) &&
+                a.XVector.IsAlmostEqualTo(b.XVector) &&
+                a.YVector.IsAlmostEqualTo(b.YVector) &&
+                a.ZVector.IsAlmostEqualTo(b.ZVector);
+        }
+
+        private static bool CompareRandomParameterLocationDistances(
+            Curve a, Curve b, double tolerance)
+        {
+            for (var i = 0; i < 10; i++)
```

Comment:
I agree 10 is too few. But, we need to balance this call being too expensive with being fool-proof. You could say the same about 10,000 points being the same. We have to start somewhere :)